presto-main/src/main/java/com/facebook/presto/execution/MemoryRevokingScheduler.java

```diff
@@ -56,17 +80,118 @@ public MemoryRevokingScheduler(LocalMemoryManager localMemoryManager, FeaturesCo
         this.memoryPools = ImmutableList.of(memoryPool);
         this.memoryRevokingThreshold = memoryRevokingThreshold;
         this.memoryRevokingTarget = memoryRevokingTarget;
+        this.potentiallyOverflowingPools = synchronizedSet(Sets.newHashSetWithExpectedSize(this.memoryPools.size()));
+    }
+
+    @PostConstruct
+    public void start()
+    {
+        memoryPools.forEach(memoryPool -> memoryPool.addListener(memoryPoolListener));
+    }
+
+    @PreDestroy
+    public void stop()
+    {
+        memoryPools.forEach(memoryPool -> memoryPool.removeListener(memoryPoolListener));
+    }
+
+    void register(Supplier<? extends Collection<SqlTask>> currentTasks, ScheduledExecutorService taskManagementExecutor)
```

Comment:
i would use currentTasksSupplier as variable name. It is harder to read it to me without the suffix.